lib/pbench/server/database/models/active_tokens.py

```diff
@@ -0,0 +1,57 @@
+import datetime
+from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
+from pbench.server.database.database import Database
+
+
+class ActiveTokens(Database.Base):
+    """Token model for storing the active auth tokens at any given time"""
+
+    __tablename__ = "active_tokens"
+    id = Column(Integer, primary_key=True, autoincrement=True)
+    token = Column(String(500), unique=True, nullable=False, index=True)
+    created = Column(DateTime, nullable=False)
+    user_id = Column(
+        Integer,
+        ForeignKey("users.id", ondelete="CASCADE"),
+        nullable=False,
+        # no need to add index=True, all FKs have indexes
+    )
+
+    def __init__(self, token):
+        self.token = token
+        self.created = datetime.datetime.now()
+
+    @staticmethod
+    def query(token):
+        # We currently only query token database with given token
+        try:
+            token_model = (
+                Database.db_session.query(ActiveTokens).filter_by(token=token).first()
+            )
+        except Exception as e:
+            raise e
+        else:
+            return token_model
+
+    @staticmethod
+    def delete(auth_token):
+        """
+        Deletes the given auth token
+        :param auth_token:
+        :return:
+        """
+        try:
+            Database.db_session.query(ActiveTokens).filter_by(token=auth_token).delete()
+            Database.db_session.commit()
+        except Exception:
+            Database.db_session.rollback()
+            raise
+
+    @staticmethod
+    def valid(auth_token):
+        # check whether auth token is in the active database
+        res = ActiveTokens.query(auth_token)
+        if res:
+            return True
+        else:
+            return False
```

Comment:
These could both be object methods instead of class methods; I like the consistency and simplicity of `token.delete()` rather than `ActiveTokens.delete(token)`. Same for the `valid` test; in both cases we know we have an `AuthTokens` object. (And not to disrupt things but, semantically, the singular `ActiveToken` would really make more sense as a class name!)

Reply:
@dbutenhof Actually we dont have the `auth_token` object all we have is an auth_token string that user passed as a header, we have to query that string auth_token in order to check if its present in the database. The only time we have an access to auth_token object is when we create one.

Reply:
Ah. Somehow I assumed we had an object here; that's a problem with Python's flexible typing. If it was Java, the method declaration would have clearly called out a "String" type.

In fact... a docstring here with parameter and result descriptions would be a great idea.

Reply:
@dbutenhof, are you wishing for a little [hint](https://docs.python.org/3/library/typing.html)?

Reply:
I've used type hinting in scripts I've written; and, yeah, the extra documentation can really help when reading through a lot of code. The versions of Python we're using are recent enough that we could and probably should start using them...